### ISSUE TITLE
Qwen2-7B-Instruct 使用3090显卡推理示例代码耗时异常！

### ISSUE THREAD

**Issue by chenin-wang** [QwenLM/Qwen2#553]
3090显卡推理耗时异常，输入【1，28】，输出【1，2076】。耗时：time_costs[0]=621.352

`
import time
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer
from transformers.generation import GenerationConfig
from transformers.trainer_utils import set_seed

seed = 1024
max_experiment_times = 1
context_length_per_experiment = 1
generate_length_per_experiment = 2048
use_flash_attn = False
set_seed(seed)

tokenizer = AutoTokenizer.from_pretrained("/data/xxx/Qwen2-7B-Instruct", trust_remote_code=True)

model = AutoModelForCausalLM.from_pretrained(
    "/data/xxx/Qwen2-7B-Instruct", 
    device_map="cuda:0", 
    trust_remote_code=True, 
    torch_dtype = "auto",
    )

config = GenerationConfig.from_pretrained("/data/xxx/Qwen2-7B-Instruct", trust_remote_code=True)
config.min_length = generate_length_per_experiment + context_length_per_experiment
config.max_new_tokens = generate_length_per_experiment

time_costs = []

prompt ="Give me an introduction to large language model."
messages = [{"role":"system","content":"You are a helpful assistant."},
           {"role": "user", "content": prompt}]

text = tokenizer.apply_chat_template(messages,tokenize = False,add_generation_prompt=True)

inputs = tokenizer([text], return_tensors='pt').to(model.device) # inputs["input_ids"].shape 【1，28】
t1 = time.time()
pred = model.generate(**inputs, generation_config=config,) # pred.shape【1，2076】
time_costs.append(time.time() - t1)
`
3090显卡耗时time_costs[0]=621.352
